contents/blog/product-engineer-vs-software-engineer.md

```diff
@@ -0,0 +1,77 @@
+
+Software engineers are more idealistic. They look for the best solution to the problem in front of them, focus on best practices, and build off prior work. Security, risk, and scaling weigh heavier in their minds.
+
+## How do they spend their time?
+
+Product engineers still spend the majority of their time coding, but they have responsibilities outside of coding that software engineers don't.
+
+Product engineers also spend their time talking to customers, digging into usage data, and researching the competitive landscape. They test other products, build prototypes, and brainstorm experiments to make their product better.
+
+They spend time communicating with users, their team, and other members of the company. They communicate what they learned in research, why their work relates to that research, and how to prioritize what to work on next. They own the product roadmap, and this comes with a responsibility to communicate it.
+
+Software engineers focus on writing code, testing it, and maintaining it. They look for ways to optimize code, improve scalability, and solve bugs. They are deeply focused on specific areas of technology, whether that is databases, data pipelines, backend APIs, or client-side app frameworks. This means researching and reading documentation, updates, and code.
+
+## Why do companies choose product versus software engineers?
+
+Product engineers are often a part of smaller teams, where they must do more themselves. Startups and earlier-stage companies that require people to do more are often looking for product engineers (even if they don’t say it). They need people who can own large decisions and ship quickly. Product engineers fit this need.
+
+Software engineers are part of larger teams, usually spread out across multiple teams. They have a team to support them, so they can specialize. Their team might include designers, product managers, and marketers. This allows them to think less about if what they are building is right, and more about the implementation.
+
+Companies choose roles and team structures based on their goals. Small companies (or innovative teams within larger companies) need people who can ideate and execute product ideas. Large and mature companies need people who can specialize in key areas of technology and optimize them.
+
+For example, when Elon Musk bought Twitter, he had the goal of making dramatic changes to the product. He needed product engineers to quickly execute these changes. He wanted proof they could do this, which is why he forced engineers to show him what code they shipped recently. He didn't need software engineers to continue to maintain or optimize the existing codebase. This led to a big shift in team structure, with a large number of software engineers (and other roles) leaving the company.
```

Comment:
```suggestion
Elon Musk's Twitter is a good example of company that might benefit from product engineers – whether he's aware of the fact or not. Elon wants to make dramatic changes to the product, but in his opinion the organization he inherited was too large and unwieldy to achieve this. In other words, he doesn't need software engineers to maintain or optimize the existing codebase; he needs product engineers who can identify problems and opportunities, and execute solutions independently. 
```

Just a slight change of emphasis as we were probably projecting a bit more intent than we could easily support.